proposals/044-integration-model.md

```diff
@@ -0,0 +1,271 @@
+
+* Multiple `IntegrationRevisions` for an `Integration`
+* Exactly one of these IntegrationRevisions can be active.
+* Users can switch `IntegrationRevisions` versions from the UI
+* `IntegrationRevisions` can be inspected from a history page.
+* Old `IntegrationRevisions` can't be changed but can be used as a blueprint for new `IntegrationRevisions`.
+* As soon as an integration is changed, a new `IntegrationRevision` is created. All changes go to this new revision until it is deployed by the user. After this, the revision gets a fixed version and cannot be changed anymore (i.e. its immutable)
+* The user must always be able to see and examine the currently active `IntegrationRevision` along with its version number.
+
+## IntegrationRevision state model
+
+An `Integration` is the domain object describing Camel Routes used for performing integration tasks. 
+Each `Integration` has properties which can be changed by a user from the UI like its name, so its mutable.
+
+The actual logic like the connectors used and the configuration is encapsulated in an `IntegrationRevision`. 
+Multiple `IntegrationRevision` can be connected to a single `Integration`.
+Each `IntegrationRevision` represents a certain version of the defined integration and can be in different states.
+
+Both objects are managed by `syndesis-rest` and `syndesis-ui`. 
+
+When a user creates an integration, an overall `Integration` object is created as well as an `IntegrationRevision` which starts its life in the state _Draft_. 
+A revision in this state has no runtime artefact attached and can be freely changed and saved by an user to the database. 
+This is typically done automatically during the UI process of creating an integration.
+Changing a draft revision is a cheap operation.
+
+After a "Publish" UI operation the current revision is then transformed into a runnable artefact. 
+This runnable artefact consists of a GitHub repo containing the code and configuration plus an OpenShift resource configuration which transforms this code into running application pods.
+Running integration revisions can be suspended and resumed without destroying the deployment.
+
+Once an integration revision is published and transfered to the state _Active_ it becomes **immutable** and can not be changed afterwards (except for state related properties).
+
+`IntegrationRevisions` can be in one of the following state:
+
+![Integration state model](images/integration-state.png "Integration State Model")
+
+| State       | Description |
+| ----------- |------------ |
+| Draft | Initial state of an integration. The `IntegrationRevision` is not yet deployed |
+| Active | `IntegrationRevision` is deployed and running |
+| Inactive | `IntegrationRevision` is deployed but is not running|
+| Undeployed | An `IntegrationRevision` has been undeployed | 
+| Error | The `IntegrationRevision` is deployed but in an error state |
+
```

Comment:
Tricky bit with the state here, `Integration` and `IntegrationRevision`, the `Integration`'s state is likely going to always be the most recent `IntegrationRevision`'s state _except_ when the most recent `IntegrationRevision` is in Draft.